src/test/java/com/google/api/codegen/discogapic/transformer/DiscoGapicParserTest.java

```diff
@@ -48,6 +48,10 @@ public void testGetQualifiedResourceIdentifier() {
                     "{project}/global/backendBuckets/{backendBucket}")
                 .toUpperCamel())
         .isEqualTo("ProjectGlobalBackendBucket");
+    Truth.assertThat(
+            DiscoGapicParser.getQualifiedResourceIdentifier("{project}/zones/{zone}/disks/{disk}")
+                .toUpperCamel())
+        .isEqualTo("ProjectZoneDisk");
```

Comment:
This extra test checks that 
`{project}/zones/{zone}/disks/{disk}`
 resolves to a different resourceName than
`{project}/zones/{zone}/disks/{resource}`
The is important because the respective gapic ResourceName classes will have a field/getter/setter name for the `disk` and `resource` field, respectively, so they need to be in separate classes.

Reply:
This is interesting - should we be saying "actually, these are the same format, so they should use the same ResourceName class, and having two ResourceName classes with the same format is invalid"? The argument would be otherwise you can have an odd situation where you want one type but you have the other. The argument against might be, if the same format is used in two different places for different purposes, then we don't want them to be interchangeable. In practice I don't know if this every happens, because resource names often map to REST URLs, so the same formats can't be used for different purposes in that way. I'm not sure what the right answer is here. @garrettjonesgoogle, thoughts?

Reply:
I think it comes down to interchangeability. @andreamlin , do you know if there are any places where both are accepted? If not, it would make sense for them to be separate. Can we check the original proto2 to see what's happening there?

It should be noted that such a situation isn't possible with One Platform APIs because our original data source for config gen has stars (e.g. https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto#L49 ), not bracketed variable names, so we guess at what the variable names should be based on the plural words. In this case, we'd only have 'projects/*/zones/*/disks/*' to go off of.

Reply:
The discovery doc will explicitly ask for the bracketed word as a parameter, e.g. method [compute.disks.setLabels](https://raw.githubusercontent.com/googleapis/discovery-artifact-manager/master/discoveries/compute.v1.json), path  `{project}/zones/{zone}/disks/{resource}/setLabels`  lists parameters `project`, `zone`, and `resource`. Also, in proto2, these bracketed words are the exact names of fields in the request object. 

I understand that this introduces a redundancy compared to One Platform APIs, where the wildcard word doesn't matter, but I would rather have this duplication of semantically-equivalent resource names than the possible inaccuracy of assuming two different path templates are the same resource name class.

Reply:
PTAL again?